### PR TITLE
fix: CI Workflows concurrency group names

### PR DESCRIPTION
**Pull request by preetsshah**
### Describe your changes:

I worked on changing the concurrency group names for a couple of workflow files - maven-postgres-tests-build-skip.yml and maven-postgres-tests-build.yml. Since these workflows had the same group name as maven-build.yml workflow. So at any given point of time, only 1 workflow can execute among the 3 (which will be the one with latest commit) and it will cancel any in-progress workflow executions (as we have set concurrency.cancel-in-progress to true). With this change, this conflict will not happen.

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: .github/workflows/maven-postgres-tests-build-skip.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ permissions:
   checks: write
 
 concurrency: 
-  group: maven-build-${{ github.head_ref || github.run_id }}
+  group: maven-postgres-tests-skip-build-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
```
Comment (akash-jain-10) [inline .github/workflows/maven-postgres-tests-build-skip.yml:35]: Let's remove the concurrency from the skip tests!

File: .github/workflows/maven-postgres-tests-build.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ permissions:
   checks: write
 
 concurrency: 
-  group: maven-build-${{ github.head_ref || github.run_id }}
+  group: maven-postgres-tests-build-${{ github.head_ref || github.run_id }}
```
Comment (akash-jain-10) [inline .github/workflows/maven-postgres-tests-build.yml:49]:
```suggestion
  group: maven-build-postgres-${{ github.head_ref || github.run_id }}
```

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: .github/workflows/maven-postgres-tests-build-skip.yml (hunk 1 above)
Comment (preetsshah) [reply .github/workflows/maven-postgres-tests-build-skip.yml]: Removed concurrency from maven-postgres-tests-build-skip.yml

File: .github/workflows/maven-postgres-tests-build.yml (hunk 2 above)
Comment (preetsshah) [reply .github/workflows/maven-postgres-tests-build.yml]: Updated this one as well.

**Pull request merged**